tests/unit/fem/test_pa_idinterp.cpp

```diff
@@ -0,0 +1,134 @@
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "catch.hpp"
+#include "mfem.hpp"
+
+using namespace mfem;
+
+double compare_pa_id_assembly(int dim, int num_elements, int order,
+                              bool transpose)
+{
+   Mesh * mesh;
+   if (num_elements == 0)
+   {
+      if (dim == 2)
+      {
+         mesh = new Mesh("../../data/star.mesh", order);
+      }
+      else
+      {
+         mesh = new Mesh("../../data/beam-hex.mesh", order);
+
+         // Transform mesh vertices to test without alignment with coordinate axes.
+         for (int i=0; i<mesh->GetNV(); ++i)
+         {
+            double *v = mesh->GetVertex(i);
+            const double yscale = 1.0 + v[1];
+            const double zscale = 1.0 + v[2];
+            v[0] *= zscale;
+            v[1] *= zscale;
+            v[2] *= yscale;
+         }
+      }
+   }
+   else
+   {
+      if (dim == 2)
+      {
+         mesh = new Mesh(num_elements, num_elements, Element::QUADRILATERAL, true);
+      }
+      else
+      {
+         mesh = new Mesh(num_elements, num_elements, num_elements,
+                         Element::HEXAHEDRON, true);
+      }
+   }
+   FiniteElementCollection *h1_fec = new H1_FECollection(order, dim);
+   FiniteElementCollection *nd_fec = new ND_FECollection(order, dim);
+   FiniteElementSpace h1_fespace(mesh, h1_fec, dim);
+   FiniteElementSpace nd_fespace(mesh, nd_fec);
+
+   DiscreteLinearOperator assembled_id(&h1_fespace, &nd_fespace);
+   assembled_id.AddDomainInterpolator(new IdentityInterpolator);
+   const int skip_zeros = 1;
+   assembled_id.Assemble(skip_zeros);
+   assembled_id.Finalize(skip_zeros);
+   const SparseMatrix& assembled_id_mat = assembled_id.SpMat();
+
+   DiscreteLinearOperator pa_id(&h1_fespace, &nd_fespace);
+   pa_id.SetAssemblyLevel(AssemblyLevel::PARTIAL);
+   pa_id.AddDomainInterpolator(new IdentityInterpolator);
+   pa_id.Assemble();
+   pa_id.Finalize();
+
+   int insize, outsize;
+   if (transpose)
+   {
+      insize = nd_fespace.GetVSize();
+      outsize = h1_fespace.GetVSize();
+   }
+   else
+   {
+      insize = h1_fespace.GetVSize();
+      outsize = nd_fespace.GetVSize();
+   }
+   Vector x(insize);
+   Vector assembled_y(outsize);
+   Vector pa_y(outsize);
+
+   x.Randomize();
+   if (transpose)
+   {
+      assembled_id_mat.BuildTranspose();
+      assembled_id_mat.MultTranspose(x, assembled_y);
+      pa_id.MultTranspose(x, pa_y);
+   }
+   else
+   {
+      assembled_id.Mult(x, assembled_y);
+      pa_id.Mult(x, pa_y);
+   }
+
+   pa_y -= assembled_y;
+   double error = pa_y.Norml2() / assembled_y.Norml2();
+   std::cout << "dim " << dim << " ne " << num_elements << " order "
+             << order;
+   if (transpose)
+   {
+      std::cout << " T";
+   }
+   std::cout << ": error in PA identity: " << error << std::endl;
```

Comment:
Personally, I don't like all the output in tests (I know a lot of other existing unit tests do this). It really pollutes the terminal, and usually isn't very informative.

I suggest using an `INFO` message here, and if you want to see all of the messages, then run the tests with flag `-s`.

Reply:
You don't see the output when you run `make test`, only when you run `./unit_tests` or similar. I thought that was intentional so users could decide how much to see, but then again I didn't know about the `INFO` option or the `-s` flag.

Reply:
Good point — I almost always run the unit tests directly rather than with `make test` (usually so I can specify a subset of tests to run so it completes faster).

Reply:
This has been changed to an `INFO` message. I was also unaware of `INFO`. I agree that using `INFO` is an improvement, but a disadvantage is that people might not know about the `-s` flag. A lot of these `catch` features are not well documented or advertised, so it can be tricky for people to follow, whereas `cout` is straightforward.

Reply:
Agreed, we could edit our README or similar to better document these features. (There is also the flag `-r compact` for less verbose `-s` output). The available options are also shown with the `-h` or `--help` flags.

The point of `INFO` rather than outputting to stdout is that `INFO` will by default only output the message on test failure. I don't think these messages are very useful otherwise.